util/containerutil/frontend_test.go

```diff
@@ -663,3 +680,11 @@ func testLogger() conslogging.ConsoleLogger {
 	logger := conslogging.Current(conslogging.NoColor, conslogging.DefaultPadding, conslogging.Info)
 	return logger.WithWriter(&logs)
 }
+
+func deterministicPortHash(inputs ...string) int {
+	h := fnv.New32a()
+	for _, s := range inputs {
+		h.Write([]byte(s))
+	}
+	return int((h.Sum32() % 64512) + 1024) // Clamp to non-root port
+}
```

Comment:
Rootless `nerdctl` cannot automatically determine ports. I was only automatically determining them to prevent port collisions on parallel tests. This spreads them out while remaining deterministic.